extensions/wikia/TemplateClassification/TemplateClassificationView.php

```diff
@@ -7,42 +7,63 @@ class View {
 	/**
 	 * Returns HTML with Template type.
 	 * If a user is logged in it returns also an entry point for edition.
-	 * @param int $articleId
+	 * @param int $wikiId
+	 * @param \Title $title
 	 * @param \User $user
 	 * @param string $fallbackMsg
 	 * @return string
 	 */
-	public function renderTemplateType( $wikiId, $articleId, $user, $fallbackMsg = '' ) {
-		$templateType = lcfirst( ( new \TemplateClassificationService() )->getType( $wikiId, $articleId ) );
+	public function renderTemplateType( $wikiId, \Title $title, $user, $fallbackMsg = '', $templateTypeLabel = null ) {
+		if ( !$user->isLoggedIn() && !$this->isTemplateClassified( $title ) ) {
+			return $fallbackMsg;
+		}
+
+		$templateType = ( new \TemplateClassificationService() )->getType( $wikiId, $title->getArticleID() );
+		// Fallback to unknown for not existent classification
+		if ( $templateType === '' ) {
+			$templateType = \TemplateClassificationService::TEMPLATE_UNKNOWN;
+		}
+
+		if ( $templateTypeLabel === null ) {
+			$templateTypeLabel = wfMessage( 'template-classification-indicator' )->plain();
+		}
 		/**
-		 * template-classification-type-unclassified
 		 * template-classification-type-infobox
 		 * template-classification-type-navbox
 		 * template-classification-type-quote
+		 * template-classification-type-media
+		 * template-classification-type-reference
+		 * template-classification-type-navigation
+		 * template-classification-type-nonarticle
+		 * template-classification-type-design
+		 * template-classification-type-unknown
+		 * template-classification-type-data
 		 */
-		$templateName = wfMessage(
-			'template-classification-indicator',
-			wfMessage( "template-classification-type-{$templateType}" )
-		)->escaped();
-		if ( $user->isLoggedIn() ) {
-			$templateName .= $this->renderEditButton();
-			return $templateName;
-		} else {
-			if ( $this->isTemplateClassified() ) {
-				return $templateName;
-			}
-			return $fallbackMsg;
+		$templateTypeMessage = wfMessage( "template-classification-type-{$templateType}" )->plain();
+
+		$editButton = flase;
+		if ( ( new Permissions() )->userCanChangeType( $user, $title ) ) {
+			$editButton = true;
 		}
+
+		return \MustacheService::getInstance()->render(
+			__DIR__ . '/templates/TemplateClassificationViewPageEntryPoint.mustache',
+			[
+				'templateTypeLabel' => $templateTypeLabel,
+				'templateType' => $templateTypeMessage,
+				'editButton' => $editButton,
+			]
+		);
 	}
 
 	/**
 	 * Renders an entry point on a template's edit page.
-	 * @param int $articleId
+	 * @param \Title $title
 	 * @param \User $user
 	 * @return string
 	 */
-	public function renderEditPageEntryPoint( $wikiId, $articleId, \User $user ) {
-		$templateType = $this->renderTemplateType( $wikiId, $articleId, $user );
+	public function renderEditPageEntryPoint( $wikiId, \Title $title, \User $user ) {
+		$templateType = $this->renderTemplateType( $wikiId, $title, $user, '', '' );
```

Comment:
Note that second parameter is different than default that's why it's passed, but maybe we should rethink that if it's unclear?